### DISCUSSION TITLE
Run step only if previous step failed

### DISCUSSION THREAD

**Discussion by obvionaoe** [community/community#26097] [Actions]
<p>Hi, I have this workflow for rolling back a kubernetes deployment if the integration tests fail:</p>
<pre><code class="lang-auto">- name: Integration Tests
      id: integrationTests
      run: |
        chmod +x gradlew
        ./gradlew integrationTest

- name: Rollback Deployment
      if: steps.integrationTests.outcome == 'failure'
      run: |
         kubectl rollout undo -n $DEPLOYMENT_NAMESPACE deployment/$DEPLOYMENT_NAME-$ENVIRONMENT
</code></pre>
<p>but it always ignores the rollback step, even if the integration tests fail… What am I missing?</p>
<p>I’ve also tried the conclusion keyword and wrapping the expression in ${{ }} but nothing works!</p>
<p>Thanks</p>

**Comment by bissim** [comment, marked as answer]
<p>Hello <a class="mention" href="/obvionaoe">@obvionaoe</a> ,<br>
You may try to add <code>continue-on-error: true</code> to ‘Integration Tests’ step: it makes the workflow continue if that step fails.</p>

**Comment by airtower-luna** [comment]
<p>Take a look at the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions">Job status check functions</a>, in particular this note:</p>
<blockquote>
<p>If your <code>if</code> expression does not contain any of the status functions it will automatically result with <code>success()</code>.</p>
</blockquote>
<p>So you should probably do something like this, to check both for failure in general and the specific step that failed:</p>
<pre><code class="lang-yaml">- name: Rollback Deployment
  if: failure() &amp;&amp; steps.integrationTests.outcome == 'failure'
  run: |
    kubectl rollout undo -n $DEPLOYMENT_NAMESPACE deployment/$DEPLOYMENT_NAME-$ENVIRONMENT
</code></pre>

**Comment by obvionaoe** [comment]
<p>Thanks a lot, that fixed it!</p>

**Comment by mmiglioranza22** [comment]
Thanks @airtower-luna, I had a similar issue and this made the trick. Cheers
